tests/test_template_miner.py

```diff
@@ -77,12 +77,145 @@ def add_and_test(msg, expected_params):
         add_and_test("hello", [])
         add_and_test("hello ABC", [])
         add_and_test("hello BCD", ["BCD"])
+        add_and_test("hello    BCD", ["BCD"])
```

Comment:
we should also test `exact_matching=false`
it can be done on this test since it passes also without exact_matching